bee-bundle/src/transaction.rs

```diff
@@ -38,30 +36,59 @@ pub enum TransactionFieldError {
     FieldDeserializationError,
 }
 
-pub trait TransactionField: Sized {
-    fn try_from_tritbuf(buffer: TritBuf) -> Result<Self, TransactionFieldError>;
-    fn from_tritbuf_unchecked(buffer: TritBuf) -> Self;
+pub trait TransactionField: Sized + TransactionFieldType {
+    type inner;
+    fn try_from_inner(buffer: Self::inner) -> Result<Self, TransactionFieldError>;
+    fn from_inner_unchecked(buffer: Self::inner) -> Self;
 
-    fn into_inner(&self) -> TritBuf<T1B1Buf>;
+    fn into_inner(&self) -> &Self::inner;
```

Comment:
`into_inner -> to_inner`

`into_*` should be reserver for owned to owned conversions. See this guide: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv